sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3186,11 +3203,19 @@ piEnqueueKernelLaunch(pi_queue Queue, pi_kernel Kernel, pi_uint32 WorkDim,
   // Lock automatically releases when this goes out of scope.
   std::lock_guard<std::mutex> lock(Queue->PiQueueMutex);
 
+  // It is only OK to batch this kernel if there is an event that can be used
+  // to release the kernel once it has been executed.  So we will not allow
+  // these to be batched, only executed immediately.
+  // Should this also force a synchronize immediately?  That would also
+  // ensure that the kernel had completed execution before it could be
+  // released, but that may really hurt performance.
+  bool BatchingThisKernelOK = (Event);
```

Comment:
At what circumstance would the SYCL RT pass a NULL Event here? Tag @romanovvlad

Reply:
Currently SYCL RT always pass non-null pointer to pi_event.

Reply:
@romanovvlad  - Seems like it would be nice to put in an assert(Event), and take out the if.  Is this true of all the various pi entry points that take Event?

Reply:
Agree, if the plugin relies on that it would be helpful to have an assert.

Reply:
@smaslov-intel - Sergey what do you think about adding an assert and getting rid of the if?

Reply:
Yes, let's do that.

Reply:
Done.